### ISSUE TITLE
P1: Chromecast Ultra often and inconsistently decides to use HD stream instead of 4K

### ISSUE THREAD

**Issue by zmarty** [YouTubeHDR/hdr_metadata#3]
**HDR test video:** https://www.youtube.com/watch?v=WW2DKBGCvEs
(but I also tried it with several other private HDR videos)

**Expected:** When we cast the video through Chromecast Ultra in the header we should see [4K] [HDR] and the content should play with high 4K quality. By [4K] [HDR] we mean the tags with white background that show up below the title of the video.

**What actually happens:** When we cast the same video multiple times through Chromecast Ultra we see [4K] [HDR] about 10% of the time and [HD] [HDR] about 90% of the time. When [4K] is displayed the video plays at high quality. When [HD] is displayed the video plays at low HD quality.

**How we repro-ed the issue:**
- We connected an i7 Windows PC to a 4K SDR screen
- We connected a Chromecast Ultra to a 4K HDR screen
- We connected both the PC and the Ultra to the same wired network with 100 Mbps up / 100 Mbps down bandwidth
- We played the video multiple times on both devices
- We observed Chromecast Ultra **only** decides to often use the HD stream instead of the 4K stream. The PC can always play the 4K stream without any issues.

**Comment by YouTubeHDR** [comment]
Thanks for reporting the issue. The YouTube player will automatically select the format played according to the throughput of the network for you. You may select the format to play manually but that my introduce re-buffering. Did you connect chrome-cast to wired network using Ethernet adaptor?

**Comment by zmarty** [comment]
Yes, as described in the item I am using a wired network and the playback is fine on PC using the same network.

**Comment by YouTubeHDR** [comment]
Thanks, we will take care of it and get back to you ASAP.

**Comment by YouTubeHDR** [comment]
Could you right click the player screen and check "stats for nerds"? It will plot the bandwidth perceived by the player.

**Comment by zmarty** [comment]
Sure, it says: 88457 Kbps

**Comment by YouTubeHDR** [comment]
We suspect that this is due to the video file is not cached in ISP. Could you watch this all the way to the end for three times and check if the 4k version is delivered or not?

**Comment by zmarty** [comment]
Well, I tried it. I can't really tell for sure if your theory is correct. For some short videos it did work. But for one particular video it did not work, and I do not know if it's a caching issue or a transcoding issue, since it's a longer video: https://www.youtube.com/watch?v=NHykpuFkKQ8

The SDR version has a 4K version but I can never get the HDR version to play in 4K.

**Comment by zmarty** [comment]
I still have this problem and users complain about this daily. I think this is also a UX problem. The resolution labels should be changed to somehow highlight both the current resolution, and the maximum resolution.

**Comment by zmarty** [comment]
Going back to the UX problem, I want to share to you one of the comments on my videos: "Why record a video in 4k only to post it 1080p?...smh﻿". I actually get a lot of these comments. All of my videos are in 4K so there is definitely a UX problem. 

If a user casts a video using Chromecast and the video is 4K, I think it is is a mistake to just show the [HD] label. You are showing the [HD] label either because of CDN issues, or because their Internet connection is bad. But that does not change the fact that the video is in fact 4K. 

There needs to be some indication of the current resolution AND the maximum resolution.
